File: code/10-state-space-search/c/permutation_demo.c
#include "state_space_search.h"
#include <stdio.h>

int main(void) {
    uint64_t perms, combos;
    ss_metrics m;
    if (ss_permutation_count(5, 4, &perms, &m) != SS_OK) return 1;
    if (ss_combination_count(5, 4, &combos, &m) != SS_OK) return 1;
    printf("P(5,4): %llu\n", (unsigned long long)perms);
    printf("C(5,4): %llu\n", (unsigned long long)combos);
    return 0;
}

File: code/10-state-space-search/python/permutation_demo.py
from state_space_search import choose_permutation, choose_combination

if __name__ == "__main__":
    perms = choose_permutation(5, 4)
    combos = choose_combination(5, 4)
    print("P(5,4):", len(perms))
    print("C(5,4):", len(combos))
